react/src/components/UserPrefModal.tsx

```diff
@@ -0,0 +1,300 @@
+import React, { useEffect } from 'react';
+import { Modal, Input, Form, Select, Divider, message, Switch } from 'antd';
+import { useTranslation } from "react-i18next";
+import { useWebComponentInfo } from './DefaultProviders';
+import { passwordPattern } from './ResetPasswordRequired';
+import { useSuspendedBackendaiClient } from "../hooks";
+import { baiSignedRequestWithPromise } from '../helper';
+import { useTanMutation } from '../hooks/reactQueryAlias';
+
+const UserPrefModal : React.FC = () => {
+  const { t } = useTranslation();
+
+  const [form] = Form.useForm();
+
+  const [messageApi, contextHolder] = message.useMessage();
+  const _showSuccessMessage = (successMessage: string) => {
+    messageApi.open({
+      type: "success",
+      content: successMessage
+    });
+  };
+  const _showErrorMessage = (errorMessage: string) => {
+    messageApi.open({
+      type: "error",
+      content: errorMessage
+    })
+  }
+
+  const { value, dispatchEvent } = useWebComponentInfo();
+  let parsedValue: {
+    isOpen: boolean;
+    full_name: string;
+    userId: string;
+    totpSupported: boolean;
+    totpActivated: boolean;
+    loggedAccount: {
+      access_key: string;
+    };
+    keyPairInfo: {
+      keypairs: [{
+        access_key: string;
+        secret_key: string;
+      }];
+    }
+  };
+  try {
+    parsedValue = JSON.parse(value || "");
+  } catch (error) {
+    parsedValue = {
+      isOpen: false,
+      full_name: "",
+      userId: "",
+      totpSupported: false,
+      totpActivated: false,
+      loggedAccount: {access_key: ""},
+      keyPairInfo: {keypairs:[{access_key: "", secret_key: ""}]}
+    };
+  };
+  const { isOpen, full_name, userId, totpSupported, totpActivated, loggedAccount, keyPairInfo } = parsedValue;
+
+  let selectOptions: any[] = [];
+  if (keyPairInfo.keypairs) {
+    for (let i=0; i < keyPairInfo.keypairs.length; i++) {
+      selectOptions.push({value: keyPairInfo.keypairs[i].secret_key, label: keyPairInfo.keypairs[i].access_key})
+    }
+  };
+
+  const baiClient = useSuspendedBackendaiClient();
+  const mutationToUpdateUserFullName = useTanMutation({
+    mutationFn: (body: {
+      full_name: string;
+      email: string
+    }) => {
+      return baiSignedRequestWithPromise({
+        method: "POST",
+        url: `/auth/update-full-name`,
+        body,
+        client: baiClient,
+      });
+    }
+  });
```

Comment:
It can be written more simply by referring to the [following](https://github.com/lablup/backend.ai-webui/blob/1debfdce78b335728bb3d553f97c865b2c17b346/react/src/components/SessionListColums/SessionInfoCell.tsx#L56-L62). (e.g. `baiClient.update_full_name(<params>)`)